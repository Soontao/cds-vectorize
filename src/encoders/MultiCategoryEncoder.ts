import CategoryEncoder from "./CategoryEncoder.js";

/**
 * @human
 */
export class MultiCategoryEncoder<T> extends CategoryEncoder<Array<T>> {
  constructor(values: Array<any>) {
    super(values, true);
  }

  features(name: string): string[] {
    return this._values.map((v) => `${name}_has_${String(v)}`);
  }
}

export default MultiCategoryEncoder;
