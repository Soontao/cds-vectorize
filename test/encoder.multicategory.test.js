import { MultiCategoryEncoder } from "../src/encoders/MultiCategoryEncoder.js";

describe('MultiCategoryEncoder Test Suite', () => {

  it('should support encode with MultiCategoryEncoder', () => {
    const e = new MultiCategoryEncoder(["A", "B", "C"])
    expect(e.features("cat_1")).toMatchSnapshot()
    expect(e.encode(['A'])).toEqual([1, 0, 0])
    expect(e.encode(['A', "B"])).toEqual([1, 1, 0])
    expect(e.encode(['A', "B", "C"])).toEqual([1, 1, 1])
  });

  it('should support decode with MultiCategoryEncoder', () => {
    const e = new MultiCategoryEncoder(["A", "B", "C"])
    expect(e.decode(['A', "B", "C"])).toEqual([])
    expect(e.decode([0, 1, 1])).toEqual(["B", "C"])
  });


});
