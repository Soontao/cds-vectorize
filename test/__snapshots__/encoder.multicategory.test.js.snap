// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiCategoryEncoder Test Suite should support encode with MultiCategoryEncoder 1`] = `
[
  "cat_1_has_A",
  "cat_1_has_B",
  "cat_1_has_C",
]
`;
